### PR TITLE
chore: set max SDK version to <3.0.0

### PR DESCRIPTION
**Pull request by chalin**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,16 +1,21 @@
 name: watcher
-version: 0.9.7+9
+version: 0.9.8
```
Comment (chalin) [inline pubspec.yaml]: @kevmoo: maybe this would be a good time to bump this up to 1.0.0?

**Comment by chalin** [comment]
cc @kwalrath @kevmoo

File: test/directory_watcher/linux_test.dart [hunk 2]
```diff
@@ -17,8 +17,7 @@ void main() {
   sharedTests();
 
   test('DirectoryWatcher creates a LinuxDirectoryWatcher on Linux', () {
-    expect(
-        new DirectoryWatcher('.'), new isInstanceOf<LinuxDirectoryWatcher>());
+    expect(new DirectoryWatcher('.'), TypeMatcher<LinuxDirectoryWatcher>());
```
Comment (munificent) [inline test/directory_watcher/linux_test.dart]: If you're going to remove one `new`, should probably remove the other too. Or just keep the `new` for now and we'll remove them all later.

Comment (chalin) [reply test/directory_watcher/linux_test.dart]: I'll add back the `new`. We can run `dartfmt --fix` later (so as to keep this PR simple).

**Pull request merged**